client/components/data/query-post-stats/index.jsx

```diff
@@ -1,80 +1,43 @@
-import { isEqual } from 'lodash';
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestPostStats } from 'calypso/state/stats/posts/actions';
 import { isRequestingPostStats } from 'calypso/state/stats/posts/selectors';
 
-class QueryPostStats extends Component {
-	static defaultProps = {
-		requestPostStats: () => {},
-		heartbeat: 0,
-	};
+function useMemoizedFields( fields ) {
+	const memoizedFields = useRef();
+	const joinedArray = fields.join();
 
-	static propTypes = {
-		siteId: PropTypes.number,
-		postId: PropTypes.number,
-		fields: PropTypes.array,
-		requestingPostStats: PropTypes.bool,
-		requestPostStats: PropTypes.func,
-		heartbeat: PropTypes.number,
-	};
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillMount() {
-		const { requestingPostStats, siteId, postId } = this.props;
-		if ( ! requestingPostStats && siteId && typeof postId !== 'undefined' ) {
-			this.requestPostStats( this.props );
-		}
-	}
-
-	componentWillUnmount() {
-		this.clearInterval();
+	if ( joinedArray !== memoizedFields.current ) {
+		memoizedFields.current = joinedArray;
 	}
 
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		const { siteId, postId, fields, heartbeat } = this.props;
-		if (
-			! ( siteId && typeof postId !== 'undefined' ) ||
-			( siteId === nextProps.siteId &&
-				postId === nextProps.postId &&
-				isEqual( fields, nextProps.fields ) &&
-				heartbeat === nextProps.heartbeat )
-		) {
-			return;
-		}
+	return memoizedFields.current;
+}
 
-		this.requestPostStats( nextProps );
+const request = ( siteId, postId, fields ) => ( dispatch, getState ) => {
+	if ( ! isRequestingPostStats( getState(), siteId, postId, fields ) ) {
+		dispatch( requestPostStats( siteId, postId, fields ) );
 	}
+};
 
-	requestPostStats( props ) {
-		const { siteId, postId, fields, heartbeat } = props;
-		props.requestPostStats( siteId, postId, fields );
-		this.clearInterval();
-		if ( heartbeat ) {
-			this.interval = setInterval( () => {
-				props.requestPostStats( siteId, postId, fields );
-			}, heartbeat );
-		}
-	}
+function QueryPostStats( { siteId, postId, fields } ) {
+	const dispatch = useDispatch();
+	const memoizedFields = useMemoizedFields( fields );
 
-	clearInterval() {
-		if ( this.interval ) {
-			clearInterval( this.interval );
+	useEffect( () => {
+		if ( siteId && postId ) {
+			dispatch( request( siteId, postId, memoizedFields.split( ',' ) ) );
```

Comment:
Well if you're content to join the array and then split it again, you could do away without a clever hook:
```js
const joinedFields = fields.join();
useEffect( () => {
  request( joinedFields.split( ',' ) );
}, [ joinedFields ] );
```
A more generic approach would be `useMemoizedFields` returning the array, and then:
```js
const memoFields = useMemoFields( fields );
useEffect( () => {
  request( memoFields );
}, [ memoFields ] );

Reply:
You're totally right. I didn't see the forest for the trees. Fixed in https://github.com/Automattic/wp-calypso/pull/60026/commits/54f448952ee939ab41faae9bfcbc13eaced06faa